hledger-lib/Hledger/Read/Common.hs

```diff
@@ -477,19 +479,17 @@ spaceandamountormissingp =
     lift $ skipSome spacenonewline
     Mixed . (:[]) <$> amountp
 
-#ifdef TESTS
-assertParseEqual' :: (Show a, Eq a) => (Either ParseError a) -> a -> Assertion
+assertParseEqual' :: (Show a, Eq a) => (Either (ParseError Char Void) a) -> a -> Assertion
 assertParseEqual' parse expected = either (assertFailure.show) (`is'` expected) parse
```

Comment:
The problem is that parseWithState now returns `Monad m => m (Either (ParseError Char Void) a)` instead of just `Either (ParseError Char Void) a`. Fixing the monad as an `Identity`, and unwrapping it simply with `runIdentity` fixes the problem.
```
assertParseEqual' ::
     (Show a, Eq a)
  => Identity (Either (ParseError Char CustomErr) a)
  -> a
  -> Assertion
assertParseEqual' parse expected = either (assertFailure.show) (`is'` expected) (runIdentity parse)
```